frontends/p4/simplifyDefUse.cpp

```diff
@@ -614,6 +1029,18 @@ class FindUninitialized : public Inspector {
                 callee.push_back(table);
             }
         } else if (auto em = mi->to<ExternMethod>()) {
+            auto args = em->expr->arguments;
+            if (args->size() > 0) {
+                auto header = args->at(0);
+                if (em->originalExternType->getName().name == "packet_in" &&
+                    em->method->getName().name == "extract") {
+                        headerDefs->update(header->expression, true);
+                } else if (em->originalExternType->getName().name == "packet_out" &&
+                         em->method->getName().name == "emit") {
+                    reportWarningIfInvalidHeader(header->expression,
```

Comment:
why is this method special? all methods that have an 'in' argument of type containing a header should be handled in the same way.

Reply:
Yes, I agree. This method isn't much more special than other extern methods.